### PR TITLE
Add unit tests for io.restassured.RestAssured and io.restassured.specification.Argument

### PR DESCRIPTION
**Pull request by petersear**
Hi,
I've analysed your codebase and added some unit tests for the following classes:
    io.restassured.RestAssured
    io.restassured.specification.Argument
I've written the tests for these functions with the help of Diffblue Cover.
These tests should help detect regressions caused by future code changes.

### CODE REVIEW

File: rest-assured/src/test/java/io/restassured/RestAssuredTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.Timeout;
+
+public class RestAssuredTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  @Rule public Timeout globalTimeout = new Timeout(10000);
```
Comment (sudo-jaa) [inline rest-assured/src/test/java/io/restassured/RestAssuredTest.java]: Timeout and import org.junit.rules.Timeout; can be removed

File: rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import io.restassured.specification.Argument;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.Timeout;
+
+public class ArgumentTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
```
Comment (sudo-jaa) [inline rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java]: We can remove timeout, timeout rules, and expectedexception rule.

File: rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    // Arrange
+    final Argument objectUnderTest = new Argument(null);
+    final Object o = null;
+    // Act
+    final boolean retval = objectUnderTest.equals(o);
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
```
Comment (sudo-jaa) [inline rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java]: I'd think about removing a hashCode test where possible.

**Review by sudo-jaa — CHANGES_REQUESTED** [review]
Remove redundant timeout and ExpectedException rules, and hashCode test.

File: rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java (hunk 3 above)
Comment (petersear) [reply rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java]: Removed

File: rest-assured/src/test/java/io/restassured/RestAssuredTest.java (hunk 1 above)
Comment (petersear) [reply rest-assured/src/test/java/io/restassured/RestAssuredTest.java]: Thanks. Removed.

File: rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java (hunk 2 above)
Comment (petersear) [reply rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java]: Removed.

File: rest-assured/src/test/java/io/restassured/RestAssuredTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    final String userName = null;
+    final String password = "";
+    final FormAuthConfig config = null;
+    // Act
+    thrown.expect(IllegalArgumentException.class);
+    RestAssured.form(userName, password, config);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @Test
```
Comment (paulkennethkent) [inline rest-assured/src/test/java/io/restassured/RestAssuredTest.java]: I don't think this `preemptiveOutputNotNull ` test is meaningful enough to submit

File: rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  public void withArgInputZeroOutputNotNull() {
```
Comment (paulkennethkent) [inline rest-assured/src/test/java/io/restassured/specification/ArgumentTest.java]: I don't think this `withArgInputZeroOutputNotNull` test is meaningful enough to submit. If you look at what it's testing something super simple that can't really break?

**Review by paulkennethkent — CHANGES_REQUESTED** [review]

**Comment by petersear** [comment]
All comments addressed, new changes pushed.

**Review by JohnLBergqvist — APPROVED** [review]
I presume there was no `true` test-case for that last one?

**Review by sudo-jaa — APPROVED** [review]

**Comment by petersear** [comment]
Thanks. No, only three tests generated for that class.

**Pull request closed without merging**